include/quadrilateral_gimp_element.h

```diff
@@ -0,0 +1,68 @@
+#ifndef MPM_GIMP_ELEMENT_H_
+#define MPM_GIMP_ELEMENT_H_
+
+#include <exception>
+
+#include <Eigen/Dense>
+
+#include "logger.h"
+#include "quadrilateral_element.h"
+
+namespace mpm {
+
+//!   13----------12----------11----------10
+//!   |           |           |           |
+//!   |           |           |           |
+//!   |           |           |           |
+//!   |           |           |           |
+//!   |           |           |           |
+//! 		        (-1, 1)	    (1,1)
+//!   14----------3-----------2-----------9
+//!   |           |           |           |
+//!   |           |           |           |
+//!   |           |   Point   |           |
+//!   |           | location  |           |
+//!   |           |           |           |
+//!   15----------0-----------1-----------8
+//!		         (-1,-1)	    (1,-1)
+//!   |           |           |           |
+//!   |           |           |           |
+//!   |           |           |           |
+//!   |           |           |           |
+//!   |           |           |           |
+//!   4-----------5-----------6-----------7
+//!
+//! </pre>
+//! \tparam Tdim Dimension
+//! \tparam Tnfunctions Number of functions
+template <unsigned Tdim, unsigned Tnfunctions>
+class QuadrilateralGIMPElement : public QuadrilateralElement<2, 4> {
+
+ public:
+  //! Define a vector of size dimension
+  using VectorDim = Eigen::Matrix<double, Tdim, 1>;
+
+  //! constructor with number of shape functions
+  QuadrilateralGIMPElement() : QuadrilateralElement<2, 4>() {
+    static_assert(Tdim == 2, "Invalid dimension for a GIMP element");
+    static_assert((Tnfunctions == 16),
```

Comment:
This check may fail if you are inheriting from Quadrilateral, you might want to just inherit from Element